### PR TITLE
chore: test with Node.js v18 in CI

### PR DESCRIPTION
**Pull request by ariya**
Also, cache npm dependencies.

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -16,7 +16,7 @@ jobs:
 
         strategy:
             matrix:
-                node-version: [14.x, 16.14]
+                node-version: [14.x, 16.x, 18.x]
```
Comment (ariya) [inline .github/workflows/node.js.yml:19]: Also, fixes `16.14` which doesn't make sense.

Comment (sapiderman) [reply .github/workflows/node.js.yml:19]: hmm could it be due to this Fetch API?

```bash
(node:113293) ExperimentalWarning: The Fetch API is an experimental feature. This feature could change at any time
(Use `node --trace-warnings ...` to show where the warning was created)
```

Comment (sapiderman) [reply .github/workflows/node.js.yml:19]: It seems the msw is not properly supporting node v.18 yet. 
As per this open issue: https://github.com/mswjs/msw/issues/1388

Comment (sapiderman) [reply .github/workflows/node.js.yml:19]: This issue is fixed in #967... closing this PR

**Pull request closed without merging**